jvm/src/main/java/com/microsoft/bond/protocol/UnsignedHelper.java

```diff
@@ -2,29 +2,134 @@
+     * @return an int containing the unsigned 8-bit value
+     */
+    static short asUnsignedShort(byte signedValue) {
+        return (short)(signedValue & 0xFF);
+    }
+
+    /**
+     * Converts an unsigned 8-bit value (represented by the signed byte data type) to a 32-bit int value.
+     * The result represents a non-negative value within the range of 8-bit unsigned integers.
+     * @param signedValue the value, interpreted as unsigned
+     * @return an int containing the unsigned 8-bit value
+     */
+    static int asUnsignedInt(byte signedValue) {
+        return signedValue & 0xFF;
+    }
+
+    /**
+     * Converts an unsigned 16-bit value (represented by the signed short data type) to a 32-bit int value.
+     * The result represents a non-negative value within the range of 16-bit unsigned integers.
+     * @param signedValue the value, interpreted as unsigned
+     * @return an int containing the unsigned 16-bit value
+     */
+    static int asUnsignedInt(short signedValue) {
+        return signedValue & 0xFFFF;
     }
 
-    public static int asUnsignedInt(short signed) {
-        return signed & 0xFFFF;
+    /**
+     * Converts an unsigned 8-bit value (represented by the signed byte data type) to a 64-bit long value.
+     * The result represents a non-negative value within the range of 8-bit unsigned integers.
+     * @param signedValue the value, interpreted as unsigned
+     * @return an int containing the unsigned 8-bit value
+     */
+    static long asUnsignedLong(byte signedValue) {
+        return signedValue & 0xFFL;
     }
 
-    public static long asUnsignedLong(int signed) {
-        return signed & 0xFFFFFFFFL;
+    /**
+     * Converts an unsigned 16-bit value (represented by the signed short data type) to a 64-bit long value.
+     * The result represents a non-negative value within the range of 16-bit unsigned integers.
+     * @param signedValue the value, interpreted as unsigned
+     * @return an int containing the unsigned 16-bit value
+     */
+    static long asUnsignedLong(short signedValue) {
+        return signedValue & 0xFFFFL;
+    }
+
+    /**
+     * Converts an unsigned 32-bit value (represented by the signed int data type) to a 64-bit long value.
+     * The result represents a non-negative value within the range of 32-bit unsigned integers.
+     * @param signedValue the value, interpreted as unsigned
+     * @return an int containing the unsigned 32-bit value
+     */
+    static long asUnsignedLong(int signedValue) {
+        return signedValue & 0xFFFFFFFFL;
+    }
+
+    /**
+     * Converts an unsigned 8-bit value (represented by the signed byte data type) to a BigInteger value.
+     * The result represents a non-negative value within the range of 8-bit unsigned integers.
+     * @param signedValue the value, interpreted as unsigned
+     * @return a BigInteger object containing the unsigned 8-bit value
+     */
+    static BigInteger asUnsignedBigInt(byte signedValue) {
```

Comment:
As we discussed, in case we provide public helpers. The fact that the class is private doesn't preclude us from doing it in the future.

---
In reply to: [116110536](https://github.com/tstein/bond/pull/4#discussion_r116110536) [](ancestors = 116110536)